plugin/src/main/java/org/jenkinsci/plugins/casc/BaseConfigurator.java

```diff
@@ -178,7 +179,33 @@ private Class introspectActualClass(TypePair type) {
         return c;
     }
 
-    protected void configure(Mapping config, T instance) throws ConfiguratorException {
+    /**
+     * Build or identify the target component this configurator has to handle based on the provided configuration node.
+     * @param mapping
+     * @return
+     * @throws ConfiguratorException
+     */
+    protected abstract T instance(Mapping mapping) throws ConfiguratorException;
+
+    @Nonnull
+    @Override
+    public T configure(CNode c) throws ConfiguratorException {
+        final Mapping mapping = (c != null ? c.asMapping() : Mapping.EMPTY);
+        final T instance = instance(mapping);
+        configure(mapping, instance, false);
+        return instance;
+    }
+
+
+    @Override
+    public T check(CNode c) throws ConfiguratorException {
+        final Mapping mapping = (c != null ? c.asMapping() : Mapping.EMPTY);
+        final T instance = instance(mapping);
+        configure(mapping, instance, true);
+        return instance;
+    }
+
+    protected void configure(Mapping config, T instance, boolean dryrun) throws ConfiguratorException {
```

Comment:
Probably it should be final so that wrong implementations cannot escape the dryrun sandbox

Reply:
good point